tests/phpunit/tests/blocks/register.php

```diff
@@ -322,6 +322,73 @@ public function test_wrong_array_index_do_not_register_block_style_handle() {
 		$this->assertFalse( $result );
 	}
 
+	/**
+	 * @ticket 58605
+	 * @dataProvider data_register_block_style_handle_uses_correct_core_stylesheet
+	 *
+	 * @param string      $block_json_path Path to the `block.json` file, relative to ABSPATH.
+	 * @param string      $style_field     Either 'style' or 'editorStyle'.
+	 * @param string|bool $expected_path   Expected path of registered stylesheet, relative to ABSPATH.
+	 */
+	public function test_register_block_style_handle_uses_correct_core_stylesheet( $block_json_path, $style_field, $expected_path ) {
+		$metadata_file = ABSPATH . $block_json_path;
+		$metadata      = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
```

Comment:
instead of loading the metadata like this, by not use the parsed php? 

[see](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/blocks.php#L320-L323) 
```suggestion
		static $core_blocks_meta;
	if ( ! $core_blocks_meta ) {
		$core_blocks_meta = require_once ABSPATH . WPINC . '/blocks/blocks-json.php';
	}
	$metadata      = $core_blocks_meta[$block_name];
```

Reply:
I think for tests that doesn't really matter. The `block.json` file is the source of truth while `blocks-json.php` is a built file, so I think for tests relying on `block.json` is a more reliable approach.